src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeImageSelect.ts

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { getImageUrl } from 'src/analysis/modal-utils';
+import { ComputeImage, ComputeImageStore, useComputeImages } from 'src/analysis/useComputeImages';
+import { RuntimeToolLabel, runtimeTools } from 'src/analysis/utils/tool-utils';
+import { spinnerOverlay } from 'src/components/common';
+import { GetRuntimeItem } from 'src/libs/ajax/leonardo/models/runtime-models';
+
+export const defaultImageId = 'terra-jupyter-gatk';
+
+export interface GcpComputeImageSelectProps {
+  readonly setSelectedImage: (image: ComputeImage | undefined) => void;
+  readonly setIsCustomSelectedImage: (value: boolean) => void;
```

Comment:
these two setX callbacks should probably be unified into an onImageSelected event callback that returns a small argument object that has these two properties in it (image, and isCustom)